### ISSUE TITLE
P1: FTRL-Proximal + SGDClassifier/SGDRegressor on the existing partial_fit seam

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#3154]
Parent: #3146 · Tier **P1 — breadth** · Milestone **0.69.0**

## The gap

linfa ships `linfa-ftrl` (Follow The Regularized Leader), marked Tested/Benchmarked, as its partial-fit learner. aprender has no FTRL and no classical SGD linear learner.

## Why this is cheaper than it looks

The seam already exists. `crates/aprender-core/src/online/mod.rs:61` declares:

```rust
fn partial_fit(&mut self, x: &[f64], y: &[f64], learning_rate: Option<f64>) -> Result<f64>;
```

with two implementors already in the module. So this ticket is adding learners to an existing trait, not designing an online-learning architecture. That is why it sits at the bottom of P1 rather than in P0 — high value per unit of work, but small absolute value.

Note that aprender's `online/` directory is otherwise aimed at LLM post-training (DPO, RLVR, distillation, drift detection, curriculum). Classical streaming linear models are a genuinely different use case that the directory name currently over-promises.

## Scope

1. **FTRL-Proximal**: per-coordinate adaptive learning rates, L1 and L2 regularization, the `alpha`/`beta`/`lambda1`/`lambda2` parameterization from the McMahan et al. paper. Its selling point is sparse solutions on high-dimensional streaming data — that must be demonstrated, not claimed.
2. **`SGDClassifier`** with `hinge`, `log_loss`, `modified_huber`, `perceptron` losses; **`SGDRegressor`** with `squared_error`, `huber`, `epsilon_insensitive`.
3. Learning-rate schedules: `constant`, `optimal`, `invscaling`, `adaptive`.
4. Both implement the existing `partial_fit` trait AND the batch `Estimator` trait, so they work inside `Pipeline`.

## Acceptance criteria (each must be able to FAIL)

- [ ] **Streaming convergence**: fed a linearly separable stream, FTRL's running loss decreases monotonically over a windowed average, and final accuracy exceeds a threshold asserted in the test.
- [ ] **Sparsity is demonstrated**: with `lambda1 > 0` on a 10,000-feature problem where only 50 features are informative, at least 90% of learned weights are exactly zero. This is FTRL's entire reason to exist — a test that does not check it is testing the wrong thing.
- [ ] Order-independence bound: two different shuffles of the same stream converge to accuracies within a documented tolerance.
- [ ] `partial_fit` on chunks and a single `fit` on the concatenation produce comparable models, within a documented tolerance.
- [ ] Oracle parity vs `sklearn.linear_model.SGDClassifier` with matched hyperparameters and seed, accuracy within 2 points.
- [ ] Each loss function's gradient is finite-difference-checked.
- [ ] Both learners compose inside the existing `Pipeline` — asserted by a test that builds one.

**Comment by noahgift** [comment]
triage 2026-09-23 (aprender-dd): milestoned to 0.76.0, theme Classical ML + AutoML. It is the classical-ML parity work (#3146 epic, CLUS rows), which no 0.70–0.75 theme covered. Operator selection 2026-09-23 ("New 0.76 release theme", wording by the cop aprender-cf); moved out of `backlog`.
